Add logic tests to circle; run in parallel with cluster test

jordanlewis opened the pull request:
This commit adds the sql logic tests to the nightly circle ci run. They are now run in parallel with the cluster test. Artifacts for both tests are saved in Circle.

Example CircleCI output lives here: https://circleci.com/gh/jordanlewis/cockroach-prod/48

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach-prod/92)
<!-- Reviewable:end -->

tschottdorf commented on build/install_deps.sh:
```diff
@@ -18,15 +18,28 @@ if [[ ! -e ${TERRAFORM_VERSION} ]]; then
 
+COCKROACH_PATH="${GOPATH%%:*}/src/github.com/cockroachdb/cockroach"
+if [ ! -e ${COCKROACH_PATH} ]; then
+    git clone --depth 1 -q "https://github.com/cockroachdb/cockroach" ${COCKROACH_PATH}
```
"${COCKROACH_PATH}"

tschottdorf commented on build/install_deps.sh:
```diff
@@ -18,15 +18,28 @@ if [[ ! -e ${TERRAFORM_VERSION} ]]; then
+if [ ! -e ${COCKROACH_PATH} ]; then
+    git clone --depth 1 -q "https://github.com/cockroachdb/cockroach" ${COCKROACH_PATH}
+else
+    cd ${COCKROACH_PATH}
```
ditto

tschottdorf commented on build/install_deps.sh:
```diff
@@ -18,15 +18,28 @@ if [[ ! -e ${TERRAFORM_VERSION} ]]; then
-mkdir -p       "${GOPATH%%:*}/src/github.com/cockroachdb/"
-git clone --depth 1 -q "http://github.com/cockroachdb/cockroach" "${GOPATH%%:*}/src/github.com/cockroachdb/cockroach"
-ln -s          "${GOPATH%%:*}/src/github.com/cockroachdb/cockroach" ~/cockroach
+ln -s "${GOPATH%%:*}/src/github.com/cockroachdb/cockroach" ~/cockroach
```
I suppose `~` could also have whitespace in it, though I don't know whether that would cause issues here.

tschottdorf commented on scripts/run_clustertest.sh:
```diff
@@ -1,74 +1,20 @@
+#!/usr/bin/env bash
 
-set -eux
+set -ex
```
no `-u`?

tschottdorf commented on scripts/run_clustertest.sh:
```diff
@@ -1,74 +1,20 @@
-  done
-  echo ${a}
-}
+$(download_binary "cockroach/acceptance.test")
```
why is this `eval`'d?

tschottdorf commented on scripts/run_clustertest.sh:
```diff
@@ -1,74 +1,20 @@
-  set +e
-  echo "Job status: ${status}"
+LOGS_DIR="${CIRCLE_ARTIFACTS}/acceptance"
+mkdir -p ${LOGS_DIR}
```
"${LOGS_DIR}"

tschottdorf commented on scripts/run_clustertest.sh:
```diff
@@ -1,74 +1,20 @@
 
-  cd "${LOGS_DIR}"
-  pwd
+./acceptance.test -test.v -test.run FiveNodesAndWriter -test.timeout 24h -remote -nodes 1 -d 1h -key-name google_compute_engine -l $LOGS_DIR -cwd "${HOME}/cockroach/cloud/gce" > >(tee "${LOGS_DIR}/test.stdout.txt") 2> "${LOGS_DIR}/test.stderr.txt"
```
ditto (paranoid, I know)

tschottdorf commented on scripts/run_logictests.sh:
```diff
@@ -66,25 +68,34 @@ sqllogictest_sha=$(latest_sha ${BINARY_PATH})
+
+# Assume the test failed.
+status="0"
+if [ $? -eq 0 ]; then
```
are you sure `$?` isn't always zero as a result of the previous line?

tschottdorf commented on scripts/run_logictests.sh:
```diff
@@ -66,25 +68,34 @@ sqllogictest_sha=$(latest_sha ${BINARY_PATH})
+  instances=$(terraform output instance|cut -d'=' -f2|tr ',' ' ')
+  supervisor_hosts=$(echo ${instances}|fmt -1|awk '{print $1 ":9001"}'|xargs|tr ' ' ',')
+
+  summary=$(${monitor} --program=sql.test --addrs=${supervisor_hosts})
```
Are you planning to keep that test bashy like that or is there going to be more `go`? Just curious.

jordanlewis replied:
I'm not going to bother converting the current remainder of the daily tests to `go` because of their slightly odd characteristics and because there are probably better things to do right now. Their odd characteristics are that they spin up a "cluster" only for parallelism - the sql logic tests for instance don't form a real cluster, and instruct each node to run a different subset of the tests. It seems like it might be more trouble than it's worth to add sufficient scaffolding to our `go` terraform setup to make this easy.

jordanlewis commented:
Review status: 0 of 6 files reviewed at latest revision, 9 unresolved discussions.

---

*[build/install_deps.sh, line 23 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQr-r1-23:-KKjv0oeEY0c7EvWkNEZ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/build/install_deps.sh#L23)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> "${COCKROACH_PATH}"
</details>
Done.

---

*[build/install_deps.sh, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQr-r1-25:-KKjv7FYdnWgu_QHZk-a:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/build/install_deps.sh#L25)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto
</details>
Done.

---

*[build/install_deps.sh, line 30 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQr-r1-30:-KKjv9zc5k094Dud8Qei:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/build/install_deps.sh#L30)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> I suppose `~` could also have whitespace in it, though I don't know whether that would cause issues here.
</details>
Done.

---

*[scripts/run_clustertest.sh, line 3 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQu-r1-3:-KKjvAuCpG_l0mxgQvit:1758359711) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_clustertest.sh#L3)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> no `-u`?
</details>
Fixed.

---

*[scripts/run_clustertest.sh, line 10 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQu-r1-10:-KKjvKWCQWqpeI0uwYQw:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_clustertest.sh#L10)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> why is this `eval`'d?
</details>
Done.

---

*[scripts/run_clustertest.sh, line 13 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQu-r1-13:-KKjvLw6X_fZRcJBf0lX:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_clustertest.sh#L13)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> "${LOGS_DIR}"
</details>
Done.

---

*[scripts/run_clustertest.sh, line 15 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQu-r1-15:-KKjvMXk8ZWNdkXX4d2A:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_clustertest.sh#L15)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto (paranoid, I know)
</details>
Done.

---

*[scripts/run_logictests.sh, line 77 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQv-r1-77:-KKjvSjCjAQmt-9i6rnh:-116850222) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_logictests.sh#L77)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> are you sure `$?` isn't always zero as a result of the previous line?
</details>
You've got a very sharp eye Tobi! Not entirely sure, no. Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
LGTM (didn't review all the bash diffs closely). I'll leave it up to you whether you want to wait for @mberhault to take a look - you've probably talked to him about this (and there will likely be more iterations, I suppose).

tschottdorf commented:
Reviewed 6 of 6 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[scripts/run_logictests.sh, line 77 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQv-r1-77:-KKjw0gbavNcrwi68sGN:-1743572487) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_logictests.sh#L77)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> You've got a very sharp eye Tobi! Not entirely sure, no. Fixed.
</details>
Probably best to add a comment while you're at it, given that this is prone to breakage.

---

*[scripts/utils.sh, line 86 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjw4nIbdDuivwIHCmL:-KKjw4nIbdDuivwIHCmM:785379480) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/8f308b2bae0020bbab49998c8b812e7087a75f94/scripts/utils.sh#L86)):*
> ```Shell
> # create_junit_single_output takes a suite name, a test name, a time in
> # `date +%s.%N` format, a success boolean and a path, and outputs a junit
> # compatible xml file to the path with the input data.
> ```

Slight cringe here, but I suppose it'll do for the time being. Can't beat them all in the first fight.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92)*
<!-- Sent from Reviewable.io -->

jordanlewis commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[scripts/run_logictests.sh, line 77 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjtcaRt-MJVOKayiQv-r1-77:-KKjw_6uzzLBmObzY2KO:-499146009) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/98fd1d2f63d702b610aef10f1de117b0c8cefa33/scripts/run_logictests.sh#L77)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Probably best to add a comment while you're at it, given that this is prone to breakage.
</details>
Done.

---

*[scripts/utils.sh, line 86 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92#-KKjw4nIbdDuivwIHCmL:-KKjwjWTvWCetCWfb6eq:-1916715777) ([raw file](https://github.com/cockroachdb/cockroach-prod/blob/8f308b2bae0020bbab49998c8b812e7087a75f94/scripts/utils.sh#L86)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Slight cringe here, but I suppose it'll do for the time being. Can't beat them all in the first fight.
</details>
Cringing at the number of arguments and crappy method contract? Yeah... I'm happy to improve this. Do you have any suggestions?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach-prod/92)*
<!-- Sent from Reviewable.io -->

mberhault commented on scripts/run_clustertest.sh:
```diff
@@ -1,74 +1,20 @@
-  [[ $? -eq 0 ]] && status="PASSED" || status="FAILED"
-  set +e
-  echo "Job status: ${status}"
+BUCKET_NAME="cockroach"
```
I don't think you need these two lines anywhere.

jordanlewis replied:
Good point!

mberhault commented:
LGTM.
